#VRML_SIM R2021b utf8
WorldInfo {
  basicTimeStep 8
  coordinateSystem "NUE"
  CFM 1e-10
}
Viewpoint {
  orientation -0.9848754034750704 -0.17298391603522859 -0.009849082339705195 0.11549299157091826
  position 0.0019443230107620574 0.20519348159033554 0.5873821083824919
}
RectangleArena {
  wallHeight 0.01
}
TexturedBackgroundLight {
}
TexturedBackground {
}
Robot {
  translation 0 1e-05 0
  children [
    Solid {
      translation 0 0.1 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.447059 0.623529 0.811765
          }
          geometry DEF SUPPORT Box {
            size 0.5 0.2 0.1
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 5
        anchor 0 0.2 0
      }
      device [
        PositionSensor {
          name "hJointPos"
        }
        RotationalMotor {
          name "hJointMot"
        }
      ]
      endPoint Solid {
        translation 0 0.125 0.07
        rotation 0 0 1 0
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.8 0 0
            }
            geometry DEF PENDULUM Box {
              size 0.04 0.15 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE PENDULUM
        physics Physics {
        }
      }
    }
    TransmissionJoint {
      jointParameters HingeJointParameters {
        axis 0 0 5
        anchor 0 0.2 0
      }
      jointParameters2 HingeJointParameters {
        axis 0 0 1
        anchor 0 1 0
      }
      device [
        RotationalMotor {
          name "tJointMot"
        }
        PositionSensor {
          name "tJointPos"
        }
      ]
      multiplier 1
      endPoint Solid {
        translation 0 0.125 0.07
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.45098 0.823529 0.0862745
            }
            geometry USE PENDULUM
          }
        ]
        name "solid(2)"
        boundingObject USE PENDULUM
        physics Physics {
        }
      }
    }
  ]
  boundingObject Transform {
    translation 0 0.1 0
    children [
      USE SUPPORT
    ]
  }
  physics Physics {
  }
  controller "transmission_joint"
}
